src/crypto/index.ts

```diff
@@ -2738,6 +2738,8 @@ export class Crypto extends TypedEventEmitter<CryptoEvent, CryptoEventHandlerMap
         const senderId = event.getSender();
         if (!senderId || encryptionInfo.mismatchedSender) {
             // something definitely wrong is going on here
+
+            // previously: E2EState.Warning -> E2ePadlockUnverified -> Red/"Encrypted by an unverified session"
```

Comment:
these "previously" comments relate to the old logic in `EventTile` (https://github.com/matrix-org/matrix-react-sdk/blob/v3.80.1/src/components/views/rooms/EventTile.tsx#L579-L637). I found them useful for double-checking the logic.